### PR TITLE
Feat/epic autocompletion

### PR DESCRIPTION
**Pull request by zarghol**
Issue #13 
Please review only the last commit

### CODE REVIEW

File: RunningOrder/Shared/Representables/FocusableNSTextField.swift [hunk 1]
```diff
@@ -42,6 +50,8 @@ struct FocusableTextField: NSViewRepresentable {
 
     class Coordinator: NSObject, NSTextFieldDelegate {
         var parent: FocusableTextField
+        var oldFocused: Bool?
```
Comment (ghiitais) [inline RunningOrder/Shared/Representables/FocusableNSTextField.swift:63]: variable usage not clear, you can try to add some comments

File: RunningOrder/Shared/Representables/FocusableNSTextField.swift [hunk 2]
```diff
@@ -31,9 +30,18 @@ struct FocusableTextField: NSViewRepresentable {
         return textField
     }
 
-    func updateNSView(_ nsView: NSTextField, context: Context) {
+    func updateNSView(_ nsView: FocusableNSTextField, context: Context) {
         nsView.stringValue = value
         nsView.placeholderString = placeholder
+        if !isFocused && context.coordinator.oldFocused == true {
```
Comment (ghiitais) [inline RunningOrder/Shared/Representables/FocusableNSTextField.swift:45]: you can add some comment here to explain the nested conditions

File: RunningOrder/Shared/Representables/FocusableTextFieldStyle.swift [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            .overlay(RoundedRectangle(cornerRadius: 10)
+                        .strokeBorder(Color.accentColor, lineWidth: 1.0, antialiased: true)
+                        .opacity(borderOpacity)
+                        .animation(.easeInOut)
+            )
+            .focusable()
+    }
+}
+
+//extension View {
```
Comment (ghiitais) [inline RunningOrder/Shared/Representables/FocusableTextFieldStyle.swift]: why did you leave commented code ?

**Review by ghiitais — APPROVED** [review]

File: RunningOrder/Shared/Representables/FocusableTextFieldStyle.swift (hunk 3 above)
Comment (zarghol) [reply RunningOrder/Shared/Representables/FocusableTextFieldStyle.swift]: I couldn't make it work, but it is the way SwiftUI builds Styles for controls such as buttons, list, etc... so I wanted to keep it for later to re-work it, now that it is in the git history, I will delete it :)

**Review by Neckow — APPROVED** [review]

File: RunningOrder/Shared/Representables/FocusableNSTextField.swift (hunk 2 above)
Comment (zarghol) [reply RunningOrder/Shared/Representables/FocusableNSTextField.swift:45]: comments added

File: RunningOrder/Shared/Representables/FocusableNSTextField.swift (hunk 1 above)
Comment (zarghol) [reply RunningOrder/Shared/Representables/FocusableNSTextField.swift:63]: comments added

**Pull request merged**